### PR TITLE
Members only

### PR DESCRIPTION
**Pull request by prolajumokeoni**
In this project:
- Created a Post model and a Posts controller
- Users can add post and comments
- Only when signed in can users create post
- Authenticated users with devise helpers

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  def new
+    @post = Post.new
+  end
+
+  # GET /posts/1/edit
+  def edit; end
+
+  # POST /posts or /posts.json
+  def create
+    # @post = Post.new(post_params)
```
Comment (SajjadAhmad14) [inline app/controllers/posts_controller.rb]: Please remove commented code from here.

File: app/views/devise/mailer/email_changed.html.erb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @email %>!</p>
+
+<% if @resource.try(:unconfirmed_email?) %>
+  <p>We're contacting you to notify you that your email is being changed to <%= @resource.unconfirmed_email %>.</p>
+<% else %>
+  <p>We're contacting you to notify you that your email has been changed to <%= @resource.email %>.</p>
+<% end %>
```
Comment (SajjadAhmad14) [inline app/views/devise/mailer/email_changed.html.erb:7]: Please move this logic to app/helpers. [ OPTIONAL ]

File: app/views/devise/shared/_error_messages.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+                 resource: resource.class.model_name.human.downcase)
+       %>
+    </h2>
+    <ul>
+      <% resource.errors.full_messages.each do |message| %>
+        <li><%= message %></li>
+      <% end %>
+    </ul>
+  </div>
+<% end %>
```
Comment (SajjadAhmad14) [inline app/views/devise/shared/_error_messages.html.erb:15]: Please move this logic to app/helpers. [ OPTIONAL ]

File: app/views/posts/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  <% if post.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(post.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+      <ul>
+        <% post.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
```
Comment (SajjadAhmad14) [inline app/views/posts/_form.html.erb:11]: Please move this logic to app/helpers. [ OPTIONAL ]

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    <% @posts.each do |post| %>
+      <tr>
+        <td><%= post.title %></td>
+        <td><%= post.body %></td>
+        <td><%= link_to 'Show', post %></td>
+        <% if user_signed_in? %>
+        <td><%= post.user_id %></td>
+        <td><%= link_to 'Edit', edit_post_path(post) %></td>
+        <td><%= link_to 'Destroy', post, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <% end %>
```
Comment (SajjadAhmad14) [inline app/views/posts/index.html.erb]: Please move this logic to app/helpers. [ OPTIONAL ]

File: app/views/shared/_nav.html.erb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+          <%= link_to "Logout", destroy_user_session_path, method: :delete, class:"nav-link text-light fs-4" %>
+        </li>
+        <% else %>
+        <li class="nav-item">
+            <%= link_to 'Sign In', new_user_session_path, class: "nav-link fs-4"  %>
+        </li>
+        <li class="nav-item">
+            <%= link_to 'Sign Up', new_user_registration_path, class: "nav-link fs-4" %>
+        </li>
+        <% end %>
```
Comment (SajjadAhmad14) [inline app/views/shared/_nav.html.erb]: Please move this logic to app/helpers. [ OPTIONAL ]

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [x] Git flow is properly used.
- [x] README has step-by-step instructions to run the project locally.
## To Improve 🙏
- [x] Please remove commented code. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/debugging_requirement.png).
- [x] Login user should see post author name not user_id. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_see_author_requirement.png).
- [x] App should protect its routes and only expose to routes that are used.  It creates the application routes using resources :posts, only: [:new, :create, :index].
- [x] User should not be able to access posts/new route if he's not signed in.
- [x] App should use before_action to restrict access to the :new and :create methods (uses ApplicationController helper method). Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png).
- [x] Please see inline comments for reference.
# Optional
- [ ] You can deploy the app to Netlify or Heroku.
- [ ] Views should be completely separate from app logic. You can move view-related logic to app/helpers.
### Happy Coding ☺️
## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED👍️👍️👍️**

Hello @prolajumokeoni 

Great work making changes👏️👏️👏️

You can merge👍️👍️

Happy coding💻️💻️💻️

**Pull request closed without merging**